Bang/Game/DrawChecker.cs

```diff
@@ -0,0 +1,51 @@
+using Bang.Characters;
+using Bang.PlayingCards;
+
+namespace Bang.Game
+{
+    public abstract class DrawChecker
+    {
+        protected abstract bool ShouldApplyEffect(BangGameCard card);
+
+        public bool Draw(Gameplay gameplay, Character victim)
+        {
+            var firstCard = gameplay.DealCard();
+
+            bool draw = ShouldApplyEffect(firstCard);
+            
+            gameplay.Discard(firstCard);
+
+            if (victim is LuckyDuke)
+            {
+                var secondCard = gameplay.DealCard();
+
+                draw |= ShouldApplyEffect(secondCard);
```

Comment:
I am not a big fun of bitwise operators. Do you really neead it here?

Reply:
I think it's the same as 
draw = draw || ShoudApplyEffect(secondCard)

But you're right: [the logical conditional operators don't support compound assignment](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators).

Fixed